File: 01-Programming-Languages/Dart/Data-Structures/Stack/stack.dart
class Stack<T> {
  final List<T> _items = [];

  void push(T item) => _items.add(item);

  T? pop() => _items.isEmpty ? null : _items.removeLast();

  T? peek() => _items.isEmpty ? null : _items.last;

  bool get isEmpty => _items.isEmpty;

  int get size => _items.length;
}

void main() {
  final stack = Stack<int>();
  stack.push(1);
  stack.push(2);
  stack.push(3);

  print('Stack size: ${stack.size}');
  print('Popped: ${stack.pop()}');
  print('Top: ${stack.peek()}');
}

File: 01-Programming-Languages/Rust/Data-Structures/Stack/stack.rs
struct Stack {
    items: Vec<i32>,
}

impl Stack {
    fn new() -> Self {
        Stack { items: Vec::new() }
    }

    fn push(&mut self, item: i32) {
        self.items.push(item);
    }

    fn pop(&mut self) -> Option<i32> {
        self.items.pop()
    }

    fn peek(&self) -> Option<&i32> {
        self.items.last()
    }

    fn is_empty(&self) -> bool {
        self.items.is_empty()
    }

    fn size(&self) -> usize {
        self.items.len()
    }
}

fn main() {
    let mut stack = Stack::new();
    
    stack.push(1);
    stack.push(2);
    stack.push(3);

    println!("Stack size: {}", stack.size());

    if let Some(item) = stack.pop() {
        println!("Popped: {}", item);
    }

    if let Some(item) = stack.peek() {
        println!("Top: {}", item);
    }
}